ReactWindows/ReactNative/Views/TextInput/ReactPasswordBoxShadowNode.cs

```diff
@@ -0,0 +1,334 @@
+        /// <param name="uiViewOperationQueue">The UI operation queue.</param>
+        public override void OnCollectExtraUpdates(UIViewOperationQueue uiViewOperationQueue)
+        {
+            base.OnCollectExtraUpdates(uiViewOperationQueue);
+
+            if (_computedPadding != null)
+            {
+                uiViewOperationQueue.EnqueueUpdateExtraData(ReactTag, _computedPadding);
+                _computedPadding = null;
+            }
+
+            if (_jsEventCount != Unset)
+            {
+                uiViewOperationQueue.EnqueueUpdateExtraData(ReactTag, Tuple.Create(_jsEventCount, _password));
+            }
+        }
+
+        /// <summary>
+        /// Marks a node as updated.
+        /// </summary>
+        protected override void MarkUpdated()
+        {
+            base.MarkUpdated();
+            dirty();
+        }
+
+        private float[] GetDefaultPaddings()
+        {
+            // TODO: calculate dynamically
+            return new[]
+            {
+                10f,
+                3f,
+                6f,
+                5f,
+            };
+        }
+
+        private float[] GetComputedPadding()
+        {
+            return new float[]
+            {
+                GetPadding(CSSSpacingType.Left),
+                GetPadding(CSSSpacingType.Top),
+                GetPadding(CSSSpacingType.Right),
+                GetPadding(CSSSpacingType.Bottom),
+            };
+        }
+
+        private static MeasureOutput MeasureTextInput(CSSNode node, float width, CSSMeasureMode widthMode, float height, CSSMeasureMode heightMode)
+        {
+            var textInputNode = (ReactPasswordBoxShadowNode)node;
+            textInputNode._computedPadding = textInputNode.GetComputedPadding();
+
+            var borderLeftWidth = textInputNode.GetBorder(CSSSpacingType.Left);
+            var borderRightWidth = textInputNode.GetBorder(CSSSpacingType.Right);
+
+            var normalizedWidth = Math.Max(0,
+                (CSSConstants.IsUndefined(width) ? double.PositiveInfinity : width)
+                - textInputNode._computedPadding[0]
+                - textInputNode._computedPadding[2]
+                - (CSSConstants.IsUndefined(borderLeftWidth) ? 0 : borderLeftWidth)
+                - (CSSConstants.IsUndefined(borderRightWidth) ? 0 : borderRightWidth));
+            var normalizedHeight = Math.Max(0, CSSConstants.IsUndefined(height) ? double.PositiveInfinity : height);
+
+            // This is not a terribly efficient way of projecting the height of
+            // the text elements. It requires that we have access to the
+            // dispatcher in order to do measurement, which, for obvious
+            // reasons, can cause perceived performance issues as it will block
+            // the UI thread from handling other work.
+            //
+            // TODO: determine another way to measure text elements.
+            var task = DispatcherHelpers.CallOnDispatcher(() =>
```

Comment:
If you rebase, this dispatcher call is no longer needed.  All measurement runs on a UI thread.